### ISSUE TITLE
`(help for)` throws an error

### ISSUE THREAD

**Issue by zck** [arclanguage/anarki#66]
I tried to use `for` the way Arc had it, and got a helpful error message:

```
arc> (for x 1 3 (prn x))
`for` has changed in anarki. See (help for) for more information.
  context...:
    for
   /home/zck/programs/arc/anarki/ac.scm:502:0: ac-call
   /home/zck/programs/arc/anarki/ac.scm:1222:0: arc-eval
   /home/zck/programs/arc/anarki/ac.scm:1275:4
```

So I tried to look at the documentation:

```arc> (help for)
car: contract violation
  expected: pair?
  given: 'body
  context...:
   /home/zck/programs/arc/anarki/ac.scm:676:0: ar-denil-last
   /home/zck/programs/arc/anarki/ac.scm:811:10: len
   zz
    recur
   zz
   /home/zck/programs/arc/anarki/ac.scm:502:0: ac-call
   /home/zck/programs/arc/anarki/ac.scm:1275:4
```

Similarly:

```
arc> (help up)
car: contract violation
  expected: pair?
  given: 'body
  context...:
   /home/zck/programs/arc/anarki/ac.scm:811:10: len
   zz
    recur
   zz
   /home/zck/programs/arc/anarki/ac.scm:502:0: ac-call
   /home/zck/programs/arc/anarki/ac.scm:1275:4
```

Some other documentation works:

```arc> (help def)
[mac] (def name parms . body)
Defines a new function called 'name'. When called, the function runs
'body', parameterizing 'parms' with call arguments.
For more information see the tutorial: http://ycombinator.com/arc/tut.txt
Or come ask questions at http://arclanguage.org/forum
nil
arc> (help +)
[fn]  (+)
+ is not documented.
nil
```

I don't know if this is related, but it seems similar. `(type fn)` doesn't work:

```arc> (help fn) ;;this is fine
[sym] (fn params . body)
Creates an anonymous function. See the tutorial: http://ycombinator.com/arc/tut.txt
nil
arc> (type fn)  ;;this breaks
_fn: undefined;
 cannot reference an identifier before its definition
  in module: "/home/zck/programs/arc/anarki/ac.scm"
  context...:
   /home/zck/programs/arc/anarki/ac.scm:1275:4
```

I'm on version 4702cd293745fc8333fb501b45baceb53e6e1f65, committed in June. I'm using Racket 6.8.

**Comment by akkartik** [comment]
Thanks for the report! I broke it in e5ab712fbc (while trying to fix a report from jsgrahamus that `len` was slow. Now thinking about how to fix it.

**Comment by akkartik** [comment]
Fixed by commit 516a06941fe1ae3a630bd143a7842195d3153c80

**Comment by zck** [comment]
Thanks! It's working for me too. :)
